Gems/RecastNavigation/Code/Source/EditorComponents/EditorRecastNavigationMeshComponent.cpp

```diff
@@ -37,6 +42,125 @@ namespace RecastNavigation
                     ->DataElement(AZ::Edit::UIHandlers::Default, &RecastNavigationMeshComponentController::m_configuration, "Configuration", "")
                     ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::ShowChildrenOnly)
                     ;
+
+                using Config = RecastNavigationMeshConfig;
+                editContext->Class<RecastNavigationMeshConfig>("Recast Navigation Mesh Config",
+                    "[Navigation mesh configuration]")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("Game"))
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+
+                    // Agent configuration
+                    ->ClassElement(AZ::Edit::ClassElements::Group, "Agent Configuration")
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+
+                    ->DataElement(nullptr, &Config::m_agentHeight, "Agent Height",
+                        "Minimum floor to 'ceiling' height that will still allow the floor area to be considered walkable.")
+                    ->Attribute(AZ::Edit::Attributes::SoftMin, 3.f)
+                    ->Attribute(AZ::Edit::Attributes::Suffix, " world units")
+
+                    ->DataElement(nullptr, &Config::m_agentMaxClimb, "Agent Max Climb",
+                        "Maximum ledge height that is considered to still be traversable.")
+                    ->Attribute(AZ::Edit::Attributes::Min, 0.f)
+                    ->Attribute(AZ::Edit::Attributes::Suffix, " world units")
+
+                    ->DataElement(nullptr, &Config::m_agentMaxSlope, "Agent Max Slope",
+                        "The maximum slope that is considered walkable.")
+                    ->Attribute(AZ::Edit::Attributes::Min, 0.f)
+                    ->Attribute(AZ::Edit::Attributes::Max, 90.f)
+                    ->Attribute(AZ::Edit::Attributes::Suffix, " degrees")
+
+                    ->DataElement(nullptr, &Config::m_agentRadius, "Agent Radius",
+                        "The distance to erode/shrink the walkable area of the heightfield away from obstructions.")
+                    ->Attribute(AZ::Edit::Attributes::Min, 0.f)
+                    ->Attribute(AZ::Edit::Attributes::Suffix, " world units")
+
+                    // Editor-only configuration
+                    ->ClassElement(AZ::Edit::ClassElements::Group, "Editor-only Configuration")
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+
+                    ->DataElement(nullptr, &RecastNavigationMeshConfig::m_enableEditorPreview,
+                        "Editor Preview", "If enabled, frequently calculates navigation mesh and draws in the Editor viewport."
+                        "Does not affect game mode.")
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, AZ::Edit::PropertyRefreshLevels::EntireTree)
```

Comment:
Use Values or AttributesAndValues instead of EntreTree unless you're changing visibility of properties. It'll cause UI flicker/popping.

Reply:
Done!